src/pocketmine/network/mcpe/NetworkBinaryStream.php

```diff
@@ -74,6 +77,62 @@ public function putUUID(UUID $uuid) : void{
 		$this->putLInt($uuid->getPart(2));
 	}
 
+	public function getSkin() : Skin{
+		$skinId = $this->getString();
+		$skinResourcePatch = $this->getString();
+		$skinData = $this->getImage();
+		$animationCount = $this->getLInt();
+		$animations = [];
+		for($i = 0; $i < $animationCount; ++$i){
+			$animations[] = new SkinAnimation($this->getImage(), $this->getLInt(), $this->getLFloat());
+		}
+		$capeData = $this->getImage();
+		$geometryData = $this->getString();
+		$animationData = $this->getString();
+		$premium = $this->getBool();
+		$persona = $this->getBool();
+		$capeOnClassic = $this->getBool();
+		$capeId = $this->getString();
+		$fullSkinId = $this->getString();
+
+		return new Skin(
+			$skinId, $skinResourcePatch, $skinData, $animations, $capeData, $geometryData, $animationData, $premium, $persona, $capeOnClassic, $capeId
+		);
+	}
+
+	public function putSkin(Skin $skin){
+		$this->putString($skin->getSkinId());
+		$this->putString($skin->getSkinResourcePatch());
+		$this->putImage($skin->getSkinData());
+		$this->putLInt(count($skin->getAnimations()));
+		foreach($skin->getAnimations() as $animation){
+			$this->putImage($animation->getImage());
+			$this->putLInt($animation->getType());
+			$this->putLFloat($animation->getFrames());
+		}
+		$this->putImage($skin->getCapeData());
+		$this->putString($skin->getGeometryData());
+		$this->putString($skin->getAnimationData());
+		$this->putBool($skin->getPremium());
+		$this->putBool($skin->getPersona());
+		$this->putBool($skin->getCapeOnClassic());
+		$this->putString($skin->getCapeId());
+		$this->putString($skin->getFullSkinId());
+	}
+
+	public function getImage() : SerializedImage{
```

Comment:
But this does not exist in the protocol...
If you have a PC and can download the IDA, take a look at the serialize method in SerializedSkin.

Reply:
This comment is nonsense. The presence and location of this function have absolutely zero bearing on the protocol.